modules/internal/ChapelDomain.chpl

```diff
@@ -1768,13 +1793,16 @@ module ChapelDomain {
 
        .. note::
 
-          The interface and implementation is not stable and may change in the
+          The interface and implementation are not stable and may change in the
           future.
 
      :arg size: Size of the buffer in number of indices.
      :type size: int
     */
     inline proc makeIndexBuffer(size: int) {
+      if chpl_warnUnstable then compilerWarning(
+        "makeIndexBuffer() is subject to change in the future.");
```

Comment:
It's not something that's in the scope of this review but I kind of wish we had an `unstable` keyword kind of like we do for `deprecated`. It's kind of annoying that we have to manually write this if statement and warning in every place.

Pinging @lydia-duncan since I think you did the work to add the `deprecated` keyword (no good deed goes unpunished :-)). Have we thought of this before, do you have a sense of if it would be difficult to add?

At a minimum it would be nice if we just had some kind of utility function
warnUnstable(fcnName : string) that did the check and made the error for us.